ops/model.py

```diff
@@ -2598,6 +2631,15 @@ def exec(
 
         See :meth:`ops.pebble.Client.exec` for documentation of the parameters
         and return value, as well as examples.
+
+        Note that older versions of Juju do not support the ``service_content`` parameter, so if
+        the Charm is to be used on those versions, then
+        :meth:`JujuVersion.supports_exec_service_context` should be used as a guard.
+
+        Raises:
+            APIError: if an error occurred communicating with Pebble, or if the command is not
```

Comment:
I'm on the fence about this one given my comment above, but I think this one does pay its way, given the distinction between command not found (APIError) and non-zero exit code (ExecError).